### PR TITLE
[MIG] 14.0 helpdesk rating

### PR DESCRIPTION
**Pull request by PierrickBrun**
replaces https://github.com/OCA/helpdesk/pull/193

### CODE REVIEW

**Comment by ChristianSantamaria** [comment]
@PierrickBrun Tested on RunBot, it works fine but I found a bug in the qualification ticket template, the face statuses images,
one was not visible and one was in the wrong position.
I've been looking and in version 14 some rating images were renamed, now:
https://github.com/odoo/odoo/tree/14.0/addons/rating/static/src/img

imagen

**Comment by PierrickBrun** [comment]
> @PierrickBrun Tested on RunBot, it works fine but I found a bug in the qualification ticket template, the face statuses images,
> one was not visible and one was in the wrong position.
> I've been looking and in version 14 some rating images were renamed, now:
> https://github.com/odoo/odoo/tree/14.0/addons/rating/static/src/img
> 
> imagen

Thanks for the review !
It's fixed

File: helpdesk_mgmt_rating/models/helpdesk_ticket.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                        force_send=force_send,
+                    )
+
+    def rating_apply(self, rate, token=None, feedback=None, subtype=None):
+        return super().rating_apply(
+            rate,
+            token=token,
+            feedback=feedback,
+            subtype="helpdesk_mgmt_rating.mt_ticket_rating",
+        )
```
Comment (ChristianSantamaria) [inline helpdesk_mgmt_rating/models/helpdesk_ticket.py]: When giving a rating it gives an error, in v14 the rating parameter has another name.
https://github.com/odoo/odoo/blob/7015dcd56566e93418f51e8e3562f384d10b4ffa/addons/rating/models/rating_mixin.py#L172

```suggestion
    def rating_apply(self, rate, token=None, feedback=None, subtype_xmlid=None):
        return super().rating_apply(
            rate,
            token=token,
            feedback=feedback,
            subtype_xmlid="helpdesk_mgmt_rating.mt_ticket_rating",
        )
```

**Review by ChristianSantamaria — COMMENTED** [review]
Perfect, checked the images are displayed correctly. See new comment

File: helpdesk_mgmt_rating/models/helpdesk_ticket.py (hunk 1 above)
Comment (PierrickBrun) [reply helpdesk_mgmt_rating/models/helpdesk_ticket.py]: Thanks for the comment, I've fixed it and several other things

**Comment by PierrickBrun** [comment]
Hi @ChristianSantamaria I think everything is fixed now

**Comment by ChristianSantamaria** [comment]
Locally tested, LGTM 👍
Remember to change the commit name: 
"[IMP] helpdesk_mgmt_rating: pre-commit execution" to "[IMP] helpdesk_mgmt_rating: black, isort, prettier" 
to follow OCA standards and on the other hand perform a squash the fixes

**Review by ChristianSantamaria — APPROVED** [review]
👍

**Review by kevinkhao — CHANGES_REQUESTED** [review]
Works, but demo email is faulty. Rating needs to be 10/5/1. 5/3/1 is not supported

**Comment by kevinkhao** [comment]
after discussion: the opposite is true, /rate and /rating routes in mail template were the problem

**Review by kevinkhao — APPROVED** [review]

**Pull request merged**